src/Core/ContainerInterface.php

```diff
@@ -43,15 +44,15 @@ interface ContainerInterface extends PsrInterface
      * @param bool $shared Set to true to make a service shared.
      * @return \League\Container\Definition\DefinitionInterface
      */
-    public function add(string $id, $concrete = null, bool $shared = false): DefinitionInterface;
+    public function add(string $id, mixed $concrete = null, bool $shared = false): DefinitionInterface;
```

Comment:
Better to avoid making any changes to container related classes/interfaces as it will cause conflict with #15668.